pandas/core/reshape/pivot.py

```diff
@@ -537,64 +537,45 @@ def _normalize(table, normalize, margins, margins_name='All'):
             raise ValueError("Not a valid normalize argument")
 
     if margins is False:
-
         # Actual Normalizations
         normalizers = {
             'all': lambda x: x / x.sum(axis=1).sum(axis=0),
             'columns': lambda x: x / x.sum(),
             'index': lambda x: x.div(x.sum(axis=1), axis=0)
         }
 
-        normalizers[True] = normalizers['all']
-
-        try:
-            f = normalizers[normalize]
-        except KeyError:
-            raise ValueError("Not a valid normalize argument")
-
-        table = f(table)
-        table = table.fillna(0)
-
     elif margins is True:
+        # skip margin rows and/or cols for normalization
+        normalizers = {
+            'all': lambda x: x / x.iloc[:-1, :-1].sum(axis=1).sum(axis=0),
+            'columns': lambda x: x.div(x.iloc[:-1, :].sum()).iloc[:-1, :],
+            'index': lambda x: (x.div(x.iloc[:, :-1].sum(axis=1),
+                                axis=0)).iloc[:, :-1]
+        }
 
-        column_margin = table.loc[:, margins_name].drop(margins_name)
-        index_margin = table.loc[margins_name, :].drop(margins_name)
-        table = table.drop(margins_name, axis=1).drop(margins_name)
-        # to keep index and columns names
-        table_index_names = table.index.names
-        table_columns_names = table.columns.names
+    else:
+        raise ValueError("Not a valid margins argument")
 
-        # Normalize core
-        table = _normalize(table, normalize=normalize, margins=False)
+    normalizers[True] = normalizers['all']
 
-        # Fix Margins
+    try:
+        f = normalizers[normalize]
+    except KeyError:
+        raise ValueError("Not a valid normalize argument")
+
+    table = f(table)
+    table = table.fillna(0)
+
+    if margins is True:
+        # reset index to ensure default index dtype
+        if normalize == 'index':
+            colnames = table.columns.names
+            table.columns = Index(table.columns.tolist())
```

Comment:
I added these lines to make a new index from scratch, to ensure that the default index type is used. If margins is True, the index dtype always changes to 'object', because margins index value is always a string. If normalize=='index' or 'column', the margin value is removed again. As a result, an index containing only integer values has dtype object, but by default should be Int64index. I changed test_crosstab_normalize slightly and removed the dtype='object', to ensure that the validation data contains default index dtypes.